src/index.rs

```diff
@@ -288,15 +289,13 @@ impl IndexTable {
 				));
 				// We set into current the input low parts
 				let current = _mm_unpacklo_epi64(first_two, last_two);
-				let cmp = _mm_movemask_epi8(_mm_cmpeq_epi32(current, target));
+				let cmp = _mm_movemask_epi8(_mm_cmpeq_epi32(current, target)) >> (skip * 4);
```

Comment:
Is it possible `key_prefix` is 0u64 here and we still fail? (key prefix is 8 first byte from a Key so it is a 64 bit collision, cannot really say if it should be handled considering db salting it sounds like a 2^64 chance guess could be ignored).

Reply:
I mean the `>> (skip * 4)` does introduce 0 values and if target is 0 it will match (will probably create a bigger index than what could be expected though.

Reply:
Ok I misread this, works fine.